src/utils/hooks/use-generate-pathname.js

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from 'react';
+import { useHistory, generatePath } from 'react-router-dom';
+import { routes } from '../../routes.config';
+import { localStorageName } from '../../config';
+
+/**
+ * To generate different pathnames based on each action
+ * E.g.: click on a node, or focus on a modular pipeline
+ * or to reset the pathname to the main page
+ */
+export const useGeneratePathname = () => {
+  const history = useHistory();
+
+  const localStorage = window.localStorage.getItem(localStorageName);
+  const activePipeline = JSON.parse(localStorage)?.pipeline?.active;
+
+  const toFlowchartPage = useCallback(() => {
+    const url = generatePath(routes.flowchart.main);
+    history.push(url);
+  }, [history]);
+
+  const toSelectedNode = useCallback(
+    (item) => {
+      const url = generatePath(routes.flowchart.selectedNode, {
+        pipelineId: activePipeline,
+        id: item.id,
+      });
+      history.push(url);
```

Comment:
I'm noticing now that we're pushing history to the browser, which is normally fine, but when I interact with the forward or back buttons to change the URL the page doesn't update. Given your code, that makes sense, because you're only trigger that on `reload`. 

So I think we either need to allow the page to update when using the navigation buttons or not push the URLs into the browser history. I think my preference is for the former. What do you think?